modules/generator/registry/interface.go

```diff
@@ -15,9 +15,8 @@ type Counter interface {
 // Histogram
 // https://prometheus.io/docs/concepts/metric_types/#histogram
 type Histogram interface {
-	Observe(values *LabelValues, value float64)
-	// TODO support exemplars
-	//ObserveWithExemplar(lbls labels.Labels, value float64, exemplarLbls labels.Labels)
+	// Observe a datapoint with the given values. traceID will be added as exemplar.
+	Observe(values *LabelValues, value float64, traceID string)
```

Comment:
Why not use similar interface as Prometheus: `Observe` and `ObserveWithExemplar`?

Reply:
I originally had both functions but then noticed that we never use `Observe`. If we don't use it, why bother supporting it? 🤷🏻 
I'm guessing that we will always have a traceID available since every metric is based upon trace data. If at some point this isn't the case anymore we can add the original observe function again.